static/app/components/search/sources/commandSource.tsx

```diff
@@ -16,6 +20,7 @@ import {makeResolvedTs} from './utils';
 type Action = {
   action: () => void;
   description: string;
+  isVisible: boolean;
```

Comment:
Oh sorry, I was thinking this would be maybe something like
 
```suggestion
  isHidden?: (params) => boolean;
```

(also then you don't have to set the value on every thing

and then you would have this in the list

```tsx
{
  title: t('Copy Project DSN to Clipboard'),
  description: t('Copies the Project DSN to the clipboard.'),
  isHidden: params => !(params.orgId && params.projectId),
  requiresSuperuser: false,
  action: async () => {
    const api = new Client();
    const data: ProjectKey[] = await api.requestPromise(
      `/projects/${orgId}/${projectId}/keys/`
    );

    if (data.length > 0 && data[0]?.dsn?.public) {
      navigator.clipboard.writeText(data[0]?.dsn?.public);
      addSuccessMessage(t('Copied DSN to clipboard'));
    } else {
      addErrorMessage(t('No DSN found for project'));
    }
  },
```

I wish we didn't have to instantiate the `Client`, but this is tricky

Reply:
I don't know how to get `orgId` and `projectId` as variables in API url, if i don't pass `params` to a function that creates this command 🤔

I have tried a lot of things, but the best i could come up with is now pushed